server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteStoreIT.java

```diff
@@ -69,104 +72,398 @@ private IndexResponse indexSingleDoc() {
+        }
+
+        ensureRed(indices);
+        internalCluster().startDataOnlyNodes(3);
+
+        assertAcked(client().admin().indices().prepareClose(indices[0], indices[1]));
+        client().admin()
+            .cluster()
+            .restoreRemoteStore(new RestoreRemoteStoreRequest().indices(indices[0], indices[1]), PlainActionFuture.newFuture());
+        ensureGreen(indices[0], indices[1]);
+        verifyRestoredData(indicesStats.get(indices[0]), true, indices[0]);
+        verifyRestoredData(indicesStats.get(indices[1]), true, indices[1]);
+        ensureRed(indices[2], indices[3]);
+    }
+
+    /**
+     * Simulates refreshed data restored using Remote Segment Store
+     * and unrefreshed data restored using Remote Translog Store,
+     * with all remote-enabled red indices being considered for the restore
+     * except those matching the specified exclusion pattern.
+     * @throws IOException IO Exception.
+     */
+    @AwaitsFix(bugUrl = "https://github.com/opensearch-project/OpenSearch/issues/8480")
+    public void testRTSRestoreWithCommittedDataExcludeIndicesPatterns() throws IOException {
+        prepareCluster(1, 3, true, INDEX_NAMES, 1, 1);
+        String[] indices = INDEX_NAMES.split(",");
+        Map<String, Map<String, Long>> indicesStats = new HashMap<>();
+        for (String index : indices) {
+            Map<String, Long> indexStats = indexData(2, true, index);
+            indicesStats.put(index, indexStats);
+        }
+
+        for (String index : indices) {
+            if (ClusterHealthStatus.RED.equals(ensureRed(index))) {
+                continue;
+            }
+
+            if (ClusterHealthStatus.GREEN.equals(ensureRed(index))) {
+                internalCluster().stopRandomNode(InternalTestCluster.nameFilter(replicaNodeName(index)));
+            }
+
+            internalCluster().stopRandomNode(InternalTestCluster.nameFilter(primaryNodeName(index)));
+        }
+
+        ensureRed(indices);
+        internalCluster().startDataOnlyNodes(3);
+
+        assertAcked(client().admin().indices().prepareClose(indices[0], indices[1]));
+        client().admin()
+            .cluster()
+            .restoreRemoteStore(new RestoreRemoteStoreRequest().indices("*", "-remote-store-test-index-*"), PlainActionFuture.newFuture());
+        ensureGreen(indices[0], indices[1]);
+        verifyRestoredData(indicesStats.get(indices[0]), true, indices[0]);
+        verifyRestoredData(indicesStats.get(indices[1]), true, indices[1]);
+        ensureRed(indices[2], indices[3]);
+    }
+
+    /**
+     * Simulates refreshed data restored using Remote Segment Store
+     * and unrefreshed data restored using Remote Translog Store,
+     * when the index has multiple shards.
+     * @throws IOException IO Exception.
+     */
+    public void testRTSRestoreWithCommittedDataMultipleShards() throws IOException {
+        testRestoreFlow(true, 2, true, 2);
+    }
```

Comment:
we don't need a separate test for multiple shards. In existing tests only , we can make it randomized.  this will give us much better coverage .

Reply:
Good point! Updated the tests 👍